### PR TITLE
Optimize #971 get mean control

### PR DESCRIPTION
**Pull request by BeatrizPadrela**
### Linked issue

Should work alreayd for Hadamard and for the other data formats

### CODE REVIEW

File: Functions/xASL_quant_GetMeanControl.m [hunk 1]
```diff
@@ -0,0 +1,75 @@
+function imMeanControl = xASL_quant_GetMeanControl(x, imASLTimeSeries)
```
Comment (MichaelStritt) [inline Functions/xASL_quant_GetMeanControl.m:1]: Nice header 👍

**Comment by MichaelStritt** [comment]
On the first look I can't see any hard errors. If you say you already tested it on a few flavors then I'm fine, because we'll re-test everything for the release really soon anyway. Let's wait for Jan's feedback though 👍

File: Modules/SubModule_ASL/xASL_wrp_ResampleASL.m [hunk 2]
```diff
@@ -130,17 +130,10 @@ function xASL_wrp_ResampleASL(x)
...
-        ControlIm = Im(:,:,:,1);    
-        xASL_io_SaveNifti(x.P.Path_rdespiked_ASL4D, x.P.Path_mean_control, ControlIm, [], 0);
-    else
-        % Create mean control in native space
-        ControlIm = xASL_quant_GetControlLabelOrder(xASL_io_Nifti2Im(x.P.Path_rdespiked_ASL4D));
-        IM_mean = xASL_stat_MeanNan(ControlIm, 4);
-        xASL_io_SaveNifti(x.P.Path_rdespiked_ASL4D, x.P.Path_mean_control, IM_mean, [], 0);
-    end
+	% Obtain the mean control image
+	imMeanControl = xASL_quant_GetMeanControl(x, xASL_io_Nifti2Im(x.P.Path_rdespiked_ASL4D));
```
Comment (MichaelStritt) [inline Modules/SubModule_ASL/xASL_wrp_ResampleASL.m:134]: Nice use of one function as the argument of another 👍

File: Functions/xASL_quant_GetControlLabelOrder.m [hunk 3]
```diff
@@ -16,11 +15,12 @@
...
+%              with alternative control/label or label/control order
 %
 % EXAMPLE:
-% [~, ~, OrderContLabl] = xASL_quant_GetControlLabelOrder(ASLTimeSeries, x);
-% [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASLTimeSeries, x);
+% [~, ~, OrderContLabl] = xASL_quant_GetControlLabelOrder(ASLTimeSeries);
+% [ControlIm, LabelIm] = xASL_quant_GetControlLabelOrder(ASLTimeSeries);
 % 
 % __________________________________
 % Copyright (C) 2015-2019 ExploreASL
```
Comment (MichaelStritt) [inline Functions/xASL_quant_GetControlLabelOrder.m:26]: Maybe update the date while we're at it, otherwise perfect 👍

**Review by MichaelStritt — APPROVED** [review]
Let's wait for Jan's approval 👍

**Comment by jan-petr** [comment]
Some comments: NumberOfAverages might not be reliable for Hadamard, so we should really calculate the size of each Hadamard blockSize = MatrixSize*numberTEs

Then we also don't need to precalculate all the other sizes as we can then index the imASLTimeSeries directly with 1:blockSize:end

**Comment by BeatrizPadrela** [comment]
> Some comments: NumberOfAverages might not be reliable for Hadamard, so we should really calculate the size of each Hadamard blockSize = MatrixSize*numberTEs
> 
> Then we also don't need to precalculate all the other sizes as we can then index the imASLTimeSeries directly with 1:blockSize:end

Indeed! Looks great now

File: Functions/xASL_quant_GetControlLabelOrder.m (hunk 3 above)
Comment (jan-petr) [reply Functions/xASL_quant_GetControlLabelOrder.m]: Done!

**Review by jan-petr — APPROVED** [review]
OK

**Pull request merged**